File: java/LongestCommonPrefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) return "";

        int i = 0;
        String prefix = strs[0];

        do {
            for (String s : strs)
                if (i >= s.length() || s.charAt(i) != prefix.charAt(i))
                    return prefix.substring(0, i);
        } while (i++ < prefix.length());

        return prefix;
    }
}

File: php/LongestCommonPrefix.php
class Solution {

    /**
    * @param String[] $strs
    * @return String
    */
    function longestCommonPrefix($strs) {
        if (!count($strs)) return "";

        $i = 0;
        $prefix = $strs[0];

        do {
            foreach ($strs as $s)
                if ($i >= strlen($s) || $s[$i] !== $prefix[$i])
                    return substr($prefix, 0, $i);
        } while ($i++ < strlen($prefix));
    }
}
